spec/models/user_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'spec_helper'
+require 'rails_helper'
+
+describe 'User' do
+  let(:user) { build(:general_user) }
+
+  describe "valid usernames" do
+    valid_usernames = %w(
+      ryuzee
+      ryuzee-1234
+      RYUZEE
+      ryu
+    )
+    it { expect(user).to allow_value(*valid_usernames).for(:username) }
+  end
+
+  describe "invalid usernames" do
```

Comment:
__[rubocop]__
* Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)